backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/icon/BpkIcon.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.icon
+
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.layout.defaultMinSize
+import androidx.compose.material.Icon
+import androidx.compose.material.LocalContentAlpha
+import androidx.compose.material.LocalContentColor
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.utils.autoMirror
+
+enum class BpkIconSize {
+  Small,
+  Large,
+}
+
+data class BpkIcon internal constructor(
+  @DrawableRes internal val small: Int,
+  @DrawableRes internal val large: Int,
+  internal val autoMirror: Boolean,
+)
+
+@Composable
+fun BpkIcon(
```

Comment:
Maybe this should actually be the API for icons, rather than generating three icon sets (adaptive, lg, sm)

Reply:
It will not always be possible – some composables will require a specific painter to be passed in.

Reply:
It will not always be possible – some composables will require a specific painter to be passed in.